releaseNotes/MicrosoftAzureAppConfigurationAspNetCore.md

```diff
@@ -1,6 +1,16 @@
 ## Microsoft.Azure.AppConfiguration.AspNetCore
 ### [Package (NuGet)](https://www.nuget.org/packages/Microsoft.Azure.AppConfiguration.AspNetCore)
 
+### 4.0.0-preview - July 23, 2020
+* Added multi-targeting support for .NET Core 3.1 besides .NET Standard 2.0. [#173](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/173)
+
+* Updated code to leverage the new feature of dependency injection support for obtaining `IConfigurationRefresher` instances introduced in the `4.0.0-preview` version of the `Microsoft.Extensions.Configuration.AzureAppConfiguration` package.
```

Comment:
Any reasons you don't want to call out these are breaking changes explicitly?

Can we rephrase the "Updated code" to something else? You know everything is changing code. It sounds odd.

Reply:
We have multiple other breaking changes in this release and in the past. It would be confusing for readers to only see this one changed marked explicitly.

How about the following?
"Updated `IServiceCollection.AddAzureAppConfiguration()` to leverage ..."

Reply:
What other breaking changes? If we were not clear enough in the past, we should be now. It shouldn't be the reason keeping us from doing it right.



> How about the following?
> "Updated `IServiceCollection.AddAzureAppConfiguration()` to leverage ..."

No, you have details in later bullet points. This will be redundant.

Reply:
> What other breaking changes?

I believe the change to add JSON content-type support and the change to add `AppConfigurationEndpoint` to `IConfigurationRefresher` are also breaking changes. Do we want to explicitly specify it for all of these? I skipped it since I already get that information from the major version bump, so it seemed redundant.

> No, you have details in later bullet points. This will be redundant.

An alternative could be "Updated middleware to leverage ...". We already refer to the middleware in the release notes for version `2.0.0-preview-009200001-7`

Reply:
JSON content-type support falls back to string when something is wrong. It is, however, still possible a potential breaking change. If Avani can come up with scenarios, we should call them out.

The interface changes are technically breaking changes, they do not require users to update their code though. I'm fine you call it out.

Major version change does not mean everything will be broken. That's why it's important to call out what are. Does it make sense?

People will not go back and read what we mean by "middleware". We should have more explanation of what it is if we use the term.

Reply:
For users that already use a valid JSON content type for their settings, this is a breaking change. We can call that out to avoid any unexpected behavior changes.

I will check the list of issues with the tag `breaking change` in the milestone, and mark them explicitly in the release notes. This will ensure that we are consistent in our approach of deciding whether a change is breaking or not.

Reply:
Yes, JSON support is a breaking change if customers are already storing their values as valid JSON with valid JSON content type. For example, if users have this in their store:
```
key=Foo
value=[1,2,3]
contentType=application/json
```
Currently, they might be accessing this setting in their application like `config["Foo"]`. But after the new changes,  `config["Foo"]` will be null. They will now have to access `config["Foo:0"], config["Foo:1"] and config["Foo:2"]`

Reply:
What options do these users have to upgrade to the new package and let their applications continue to run as expected, other than manually updating all such settings in App Configuration?

Reply:
The only option is to update settings in AppConfiguration or the application code itself.

Reply:
How about something like this? We can add more details to the referenced issue, if needed.

* **Breaking Change :** Added JSON content-type (e.g. MIME type `application/json`) support for key-values in App Configuration. This allows primitive types, arrays, and JSON objects to be loaded properly to `IConfiguration`. Existing applications that use key-values with a valid JSON content-type may need to be updated. [#191](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/191)

Reply:
Looks good.